### DISCUSSION TITLE
v3 'issues' with phpstan

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#6958] [7. Bugs (archived)]
### Livewire version

Latest

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

I get these errors in PHPStorm. Oddly enough, phpstan is not complaining when I run it on the CLI. The code works, but we didn't have these issues in v2

```php
Livewire::actingAs($admin)
    ->test(Edit::class, [null])
```

Error: `phpstan: Call to an undefined method Livewire\Features\SupportTesting\Testable::test().`

```php
Livewire::test(Edit::class, [null])
    ->assertSuccessful()
    ->set('foo', 'bar')
```

Error: `phpstan: Call to an undefined method Illuminate\Testing\TestResponse::set().`.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by N1ebieski** [comment]
I avoid this error using:

```$this->actingAs($admin);```

**Reply by ju5t** [reply]
Yea you can avoid it, but it's a supported feature in v3, so it shouldn't really give these warnings. It didn't in v2.

**Comment by DanielGSoftware** [comment]
We run into the same issue. Interesting how it doesn't seem like a lot of other people do since this thread is from last year October.

**Comment by Luke-Shepp** [comment]
Got the same, did anyone find a solution/workaround?

```php
Livewire::test(SomeComponent::class, compact('userId'))
    ->assertSuccessful()
    ->set('someProperty', 'someValue);
```

```
 ------ --------------------------------------------------------------------- 
  27     Call to an undefined method Illuminate\Testing\TestResponse::set().  
  39     Call to an undefined method Illuminate\Testing\TestResponse::set().  
  51     Call to an undefined method Illuminate\Testing\TestResponse::set().  
 ------ --------------------------------------------------------------------- 
```

**Comment by justsanjit** [comment]
I am using v4 and running into the same issue

```php
Livewire::actingAs($user)
    ->test(MyComponent::class)
    ->assertStatus(200)
```

and the error is 

```
Unable to resolve the template type TComponent in call to method Livewire\LivewireManager::test()                                       
 🪪  argument.templateType                                              
 💡  See:                                                               
 https://phpstan.org/blog/solving-phpstan-error-unable-to-resolve-temp  
 late-type              
```
